sdk/communication/Azure.Communication.Chat/tests/ChatClients/ChatClientsTest.cs

```diff
@@ -118,25 +116,49 @@ public void ThreadCGUD_MemberAUR_MessageGSU_NotificationT()
             var getMessagesCountThread2 = messagesThread2.Count();
 
             # region Pagination assertions
-            Pageable<ChatMessage> messagesPaginationTest = chatThreadClient.GetMessages();
+            Pageable<ChatMessage> allMessages = chatThreadClient.GetMessages();
             string? continuationToken = null;
             var expectedPageSize = 2;
             var messagesTotal = 0;
-            var messagesCounter = 0;
+            var actualMessagesPerPage = 0;
+            var sentTextMessageIndex = messagesMapping.Length;
+            DateTimeOffset? previousCreatedOn = DateTimeOffset.MaxValue;
             ulong previousId = ulong.MaxValue;
-            foreach (Page<ChatMessage> messagesPage in messagesPaginationTest.AsPages(continuationToken, expectedPageSize))
+            foreach (Page<ChatMessage> messagesPage in allMessages.AsPages(continuationToken, expectedPageSize))
             {
-                (messagesCounter, continuationToken) = AssertPageableChatMessages(messagesPage, expectedPageSize, messagesMapping);
-                messagesTotal+= messagesCounter;
-
-                //Make sure expected order is ok
+                actualMessagesPerPage = messagesPage.Values.Count;
                 foreach (ChatMessage chatMessage in messagesPage.Values)
                 {
+                    if (chatMessage.Type == "Text")
```

Comment:
The sorting logic should be only based on the messages (type == "text") and as you mentioned in the other comment the order is inverse. So you could do
``` C#
var receivedMessages = messagesPage
    .Values
    .Where(x => x.Type == "Text")
    .Reverse()
    .ToArray();

for (var i = 0; i < receivedMessages.Length; i++)
{
    Assert.AreEqual(sentMessages[i].Id, receivedMessages[i].Id);
    Assert.AreEqual(sentMessages[i].Content, receivedMessages[i].Content);
}
```
Then it is a lot simpler and easier to undrestand.

Reply:
The intend of this part of the test is to make sure the paging logic we implemented on the sdk does not alter the order how the items are received from the API,  that is why I am not touching the collection with linq queries, sort or reversing statements, and just iterating them. With the dates and Id, I also want to make sure other types (UpdateTopic/AddMember) messages are in the right place. Will ping you to see if we want to add additional coverage and filter by type = text explicitly ?